### ISSUE TITLE
A NoMethodError occurred in surveyor#update

### ISSUE THREAD

**Issue by pfriedman** [NUARIG/surveyor#297]
This error happens often when running the surveys in NCS.

```
A NoMethodError occurred in surveyor#update:

undefined method `<' for nil:NilClass
surveyor (0.22.0) lib/surveyor/models/dependency_condition_methods.rb:52:in `send'

surveyor (0.22.0) lib/surveyor/models/dependency_condition_methods.rb:52:in `send'
surveyor (0.22.0) lib/surveyor/models/dependency_condition_methods.rb:52:in `is_met?'
surveyor (0.22.0) lib/surveyor/models/dependency_condition_methods.rb:37:in `to_hash'
surveyor (0.22.0) lib/surveyor/models/dependency_methods.rb:52:in `conditions_hash'
```

Clearly response.as(klass) returns nil, but not certain yet what response or klass is at this point. 

I should note that this does not affect the user entering in the responses for the survey. In fact if we did not get these exceptions emailed we would not have known about the error. Though the emails came quickly, probably occurring several times on the same survey section.

**Comment by rsutphin** [comment]
The immediate cause of one example of this exception is in a response to the following question:

``` ruby
q_HEIGHT_FT "Portion of height in whole feet (e.g., 5)",
:pick=>:one
a :integer
a_neg_1 "Refused"
a_neg_2 "Don't know"
```

The survey also contains the following dependent label:

``` ruby
label "Provided value is outside of the suggested range (4 to 7 feet). This value is admissible, but you may wish to verify."
dependency :rule=>"A or B"
condition_A :q_HEIGHT_FT, "<", {:integer_value => "4"}
condition_B :q_HEIGHT_FT, ">", {:integer_value => "7"}
```

The response set whose dependencies are being evaluated contains a response whose designated answer is the free-entry integer option, but whose `integer_value` is `nil`. When the dependency condition is being evaluated (in `DependencyConditionMethods#is_met?`), this results in `:<` being invoked on
`nil`, throwing this exception.

The response set in this case actually has 3 responses to the `HEIGHT_FT` question, even though it is not a repeatable question. This seems like at least part of the problem -- the dependency is being arbitrarily evaluated on the first response to the question and the first response happens to have this issue. A good long term goal would be for surveyor to consolidate multiple responses for non-repeatable questions. For the short term, perhaps surveyor should evaluate the condition on each response (starting from the most recently created one) until one of them does not throw an exception.

**Comment by rsutphin** [comment]
While we now have a version of Surveyor that does not create duplicates, we are still seeing cases where a response exists such that the result of `response.as(response.answer.response_class)` is `nil`. (Examining `ActsAsResponse`, there are actually times when it explicitly returns `nil`.)

So, it seems like the underlying bug here is that dependency evaluation is assuming that `response.as(...)` is never `nil` when it actually can be.

**Comment by yoon** [comment]
Closed by ca2c90a
